modules/tts/tts-adapters/tts-adapter-cereproc/src/main/java/org/daisy/pipeline/tts/cereproc/impl/CereProcEngine.java

```diff
@@ -202,4 +232,72 @@ public TTSResource allocateThreadResources() throws SynthesisException, Interrup
 		}
 		return result;
 	}
+
+	/**
+	 * Function to transform ssml to String and also perform Regex rules and char substitutions.
+	 *
+	 * @param
+	 * @return
+	 * @throws SaxonApiException
+	 */
+		public String transformSSML(XdmNode ssmlIn, Voice v) throws SynthesisException, SaxonApiException, XMLStreamException {
+
+		    List<XdmItem> ssmlProcessed = new ArrayList<>();
+
+			Configuration conf = ssmlIn.getUnderlyingNode().getConfiguration();
+
+			XMLStreamReader reader = new SaxonInputValue(ssmlIn, conf).asXMLStreamReader();
+			XMLStreamWriter writer = new SaxonOutputValue(item -> {
+				if (item instanceof XdmNode) {
+					ssmlProcessed.add(item);
+				} else {
+					throw new RuntimeException(); // should not happen
+				}
+			}, conf).asXMLStreamWriter();
+
+			performSubstitutionRules(reader, writer, v.getLocale().get().getLanguage());
+
+			if ( ssmlProcessed.size() != 1) {
+				throw new RuntimeException("Something went wrong");
+			}
+			if (!(ssmlProcessed.get(0) instanceof XdmNode)) {
+				throw new RuntimeException("Incorrect type");
+			}
+
+			XdmNode ssmlOut = (XdmNode) ssmlProcessed.get(0);
+			ThreadUnsafeXslTransformer transformer = getTransformer(ssmlOut.getUnderlyingNode().getConfiguration());
+			return transformer.transformToString(ssmlOut);
```

Comment:
Discovered another issue: the voice name should be passed here as a XSLT parameter called "voice".

Reply:
Ah, good you noticed that. I will fix it

Reply:
Added the parameter now